src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs

```diff
@@ -227,6 +232,41 @@ await requestContext.SendResult(new SchemaCompareOptionsResult()
             }
         }
 
+        /// <summary>
+        /// Handles schema compare open SCMP request
+        /// </summary>
+        /// <returns></returns>
+        public async Task HandleSchemaCompareOpenScmpRequest(SchemaCompareOpenScmpParams parameters, RequestContext<SchemaCompareOpenScmpResult> requestContext)
+        {
+            try
+            {
+                CurrentSchemaCompareTask = Task.Run(async () =>
+                {
+                    SchemaCompareOpenScmpOperation operation = null;
+
+                    try
+                    {
+                        operation = new SchemaCompareOpenScmpOperation(parameters);
+                        operation.Execute(TaskExecutionMode.Execute);
```

Comment:
Why make async methods with the operation is sync? Shouldn't this be async?

Reply:
Hmmm. Blocking the json/rpc channel seems like a bad idea as no other request can be serviced, right? I guess the question is why makes this method async in the first place? But if this is existing pattern that's used elsewhere, then I guess it makes sense to be consistent.